### ISSUE TITLE
Transition don't work when triggered by component mount

### ISSUE THREAD

**Issue by grigory-lapshin** [software-mansion/react-native-reanimated#631]
Im trying to use Transition API for `FlatList`. My `FlatList` has `ListEmptyComponent` and it is just several rows of text.

So have wrapped `FlatList` in `Transitioning.View` and add conditional rendering like this:

```
<Transitioning.View ref={this.ref} transition={transition}  style={[styles.list]}>
     {this.state.isOpen && (
       <FlatList
         data={[]}
         renderItem={() => {}}
         style={[styles.list]}
         ListEmptyComponent={this.renderEmpty}
       />
     )}
</Transitioning.View>
```

I have tried to change `this.state.isOpen` and trigger `animateNextTransition()` on mount. And it is working in iOS, but not anywhere else.

I have tried to use `useEffect` and `useLayoutEffect` and it works in simulators and on Nexus 4. It still don't works on my own Pixel 3a. Pixel 4 and some Samsung with Android 10 are fine, but Pixel 3a is special. 

And animation itself isn't right on Android. Items don't propagate one after another but move altogether.

I made snacks:
With class component and use of `componentDidMount`:
https://snack.expo.io/@grigory.lapshin/transitions-bug

With functional component and use of `useEffect`:
https://snack.expo.io/@grigory.lapshin/transition-bug-with-hooks

**Comment by fenosoa** [comment]
It's working but due to lifecycle, the result is not well calculated.
So you have to delay the animation start inside componendDidMount.
You don't have to import setTimeout as it's a javascript native function.

`componentDidMount(){
    setTimeout(this.animateNextTransition,
      200
    )
}

animateNextTransition = () => {
    this.transitionView.animateNextTransition();
}

refTransitionView = view => {
    this.transitionView = view
  }

render() {
const transition = (
      <Transition.Sequence>        
        <Transition.Together>
          <Transition.In
            type="slide-bottom"
            durationMs={400}
            interpolation="easeOut"
            propagation="bottom"
          />          
        </Transition.Together>
      </Transition.Sequence>
    );
return (
<Transitioning.View ref={this.refTransitionView} transition={transition}  style={[styles.list]}>
     {this.state.isOpen && (
       <FlatList
         data={[]}
         renderItem={() => {}}
         style={[styles.list]}
         ListEmptyComponent={this.renderEmpty}
       />
     )}
</Transitioning.View>
)}`

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
